test/test_prototype_transforms.py

```diff
@@ -152,3 +152,49 @@ def test_normalize(self, transform, input):
     )
     def test_random_resized_crop(self, transform, input):
         transform(input)
+
+
+class TestRandomHorizontalFlip:
+    def input_tensor(self, dtype: torch.dtype = torch.float32) -> torch.Tensor:
+        return torch.tensor([[[0, 1], [0, 1]], [[1, 0], [1, 0]]], dtype=dtype)
+
+    def expected_tensor(self, dtype: torch.dtype = torch.float32) -> torch.Tensor:
+        return torch.tensor([[[1, 0], [1, 0]], [[0, 1], [0, 1]]], dtype=dtype)
```

Comment:
I'm not sure we need specific values here. I think we should be good to have random image inputs and use [`torch.flip`](https://pytorch.org/docs/master/generated/torch.flip.html). I'm aware that this is the only call in the kernel

https://github.com/pytorch/vision/blob/a8bde78130fd8c956780d85693d0f51912013732/torchvision/transforms/functional_tensor.py#L126-L129

but this is also what I would use to produce a expected output if I didn't know the internals of the kernel. cc @NicolasHug for an opinion

Reply:
I used specific values to create the expecting one. I preferred to not use the `torch.flip` to not "mock" the internals of the transformation.

Reply:
> cc @NicolasHug for an opinion

No strong opinion on my side for this specific transform. In some cases it's valuable to have a simple implementation of the transform that we're testing - it helps understanding what's going on with simpler code, and we can call it on arbitrary input. In the case of `flip` the transform is very elementary and doesn't need much explaining anyway.

If we're confident that this hard-coded input covers all of what we might want to test against, then that's fine.